fec_eregs/docs/Team Practices.md

```diff
@@ -0,0 +1,143 @@
+* Decide that too many dependencies/unknowns are present and move the story to the next grooming
+
+## Daily Standups
+On days without planning or demos, the team will have a standup.  This meeting will be at most 15 minutes in length via a video call.
+Each member will provide an update on what they completed yesterday, plan for today, and highlight any blockers.
+Being a remote team, we also will update tasks with comments; however, a daily real-time meeting will help with getting team updates faster.
+
+The meeting will be from 4:30 - 4:45 EST.
+
+## Sprint Demos
+Sprint demos will be an opportunity for the team to showcase the hard work that has been completed during the sprint.
+This will be held via a video conference where members of the team can screen share the work they completed.
+Sprint demos will be open to folks outside team; particularly the FEC, our partner agency.  Sprint demos will be also open to anyone within 18F.
+
+## Retrospectives
+Retros are a forum private to the team where we will review what went well, and what did not.
+This will be a 30 minute meeting, and will be private to our team.  On request it can also be facilitated with the product steward not in attendance.
+Will be split up into the following:
+
+5 minutes : team will post in mural.ly stickies that represent:
+* What went well
+* What did not go well
+* Vote from 1-5 on how the sprint went (1 --> Poor, 5--> Excellent)
+
+5 min : moderator will group stickies together prior to team discussion
+
+10 - 15 min : open discussion on stickies
+
+5-10 min : team votes *on up to two* items to focus on for the next iteration.  There will be a lot of improvement opportunities we will not get to; but if they continue to be important to the team they will come up again in subsequent retros and get voted in.
+
+# Additional Principles
+
+## Continuous Integration
+Multiple 18F product teams will be working on a shared set of software assets.
+From Notice and Comment working out of the shared eRegs repos to FEC-data working out of beta.fec.gov.  There is likely more overlap on the former, but the latter is an asset that is also live in prod (as a beta).
+
+Therefore it is necessary to ensure that code changes preserve current functionality available to the public.  The team will invest in strengthening our automation and continous integration as we develop this product.
+
+### Smoke Tests
+Smoke tests are run on every pull request.  They are intended to validate that critical functionality is kept intact after the pull request is merged.  Smoke tests should be part of the build process and by design should complete within 5 minutes (at the very latest).  If a PR results in a failed smoke test, then the responsible dev should either address failing test or revert their code **before leaving their workstation**.  This ensures that any pull / sync from master into a branch will contain stable code.
+
+### Daily Dev Builds
```

Comment:
Looking to differentiate between tests that check for any "criticical/showstopper" breakage that execute within seconds/minutes so that a developer knows to either not merge the pull request (or rollback).  Loosely defining critical/showstoppers are those breakages that not only would prevent us from deploying code publicly, but also effectively halt development itself.

The daily builds and regression allow a framework to build out tests that could cover less severe (minor) breakages, more user paths, etc.

At the onset, will have little differences (in terms of execution) between smoke and daily regression because we're a pretty small team.  But having this framework would enable a larger team (that ma have a dedicated QA team) build out a pretty comprehensive regression test (that could take hours to execute because they are running non-intrusive tests).

Hmm, now that differentiation is brought forth, we should just start with a single test type ("Regression Tests") and at a later time see if we need to split them up into smoke vs. daily regression.